### PR TITLE
Change compactor policy from loading all binlogs to loading serially …

### PR DESCRIPTION
**Pull request by zhagnlu**
#19073 
Signed-off-by: zhagnlu <lu.zhang@zilliz.com>

### CODE REVIEW

File: internal/datacoord/compaction_trigger.go [hunk 1]
```diff
@@ -228,6 +233,28 @@ func getPlanIDs(plans []*datapb.CompactionPlan) []int64 {
...
+	resp, err := t.indexCoord.DescribeIndex(ctx, &indexpb.DescribeIndexRequest{
+		CollectionID: collectionID,
+		IndexName:    "",
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, indexInfo := range resp.IndexInfos {
+		if indexInfo.IndexName == diskAnnIndexName {
```
Comment (xige-16) [inline internal/datacoord/compaction_trigger.go:249]:
```	for _, indexInfo := range resp.IndexInfos {
		indexParamsMap := funcutil.KeyValuePair2Map(indexInfo.IndexParams)
		if indexType, ok := indexParamsMap["index_type"]; ok {
			if indexType == indexparamcheck.IndexDISKANN {
				for _, segment := range segments {
					segment.MaxRowNum = int64(Params.DataCoordCfg.DiskSegmentMaxSize)
				}
			}
		}
	}
```

File: internal/datanode/binlog_io.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ type uploader interface {
 	// errUploadToBlobStorage is returned if ctx is canceled from outside while a uploading is inprogress.
 	// Beware of the ctx here, if no timeout or cancel is applied to this ctx, this uploading may retry forever.
 	upload(ctx context.Context, segID, partID UniqueID, iData []*InsertData, segStats []byte, dData *DeleteData, meta *etcdpb.CollectionMeta) (*segPaths, error)
```
Comment (xige-16) [inline internal/datanode/binlog_io.go:54]: can removed?

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]
Compaction costs info is important, please add time elapse logs for each progress, for instance, io-upload-total-elapse, merge-total-elapse, io-download-total-elapse, etc.

**Review by congqixia — APPROVED** [review]
/lgtm

File: internal/datanode/compactor.go [hunk 3]
```diff
@@ -218,19 +272,32 @@ func (t *compactionTask) merge(mergeItr iterator, delta map[interface{}]Timestam
...
+			}
+			insertField2Path[fID] = tmpBinlog
+		}
+	}
+
+	// get pkID, pkType, dim
+	for _, fs := range meta.GetSchema().GetFields() {
 		fID2Type[fs.GetFieldID()] = fs.GetDataType()
-		if fs.GetIsPrimaryKey() {
+		if fs.GetIsPrimaryKey() && fs.GetFieldID() >= 100 && typeutil.IsPrimaryFieldType(fs.GetDataType()) {
```
Comment (czs007) [inline internal/datanode/compactor.go:290]: better to use common.StartOfUserFieldID

Comment (czs007) [reply internal/datanode/compactor.go:290]: you can modify this in next pr

**Pull request merged**